utils/model_uploader/sparse_model_autotracing.py

```diff
@@ -0,0 +1,465 @@
+
+    # 4.) Preview model config
+    print(f"\n+++++ {model_format} Model Config +++++\n")
+    with open(model_config_path, "r") as f:
+        model_config = json.load(f)
+        print(json.dumps(model_config, indent=4))
+    print("\n+++++++++++++++++++++++++++++++++++++++\n")
+
+    # 5.) Return model_path & model_config_path for model registration
+    return model_path, model_config_path
+
+
+def register_and_deploy_sparse_encoding_model(
+    ml_client: "MLCommonClient",
+    model_path: str,
+    model_config_path: str,
+    model_format: str,
+    texts: list[str],
+) -> list:
+    encoding_datas = None
+    model_id = autotracing_utils.register_and_deploy_model(
+        ml_client, model_format, model_path, model_config_path
+    )
+    autotracing_utils.check_model_status(ml_client, model_id, model_format)
+    try:
+        encoding_output = ml_client.generate_sparse_encoding(model_id, texts)
+
+        encoding_datas = [
+            encoding_output["inference_results"][i]["output"][0]["dataAsMap"][
+                "response"
+            ][0]
+            for i in range(len(texts))
+        ]
+
+    except Exception as e:
+        assert (
+            False
+        ), f"Raised Exception in generating sparse encoding with {model_format} model: {e}"
+    autotracing_utils.undeploy_model(ml_client, model_id, model_format)
+    autotracing_utils.delete_model(ml_client, model_id, model_format)
+    return encoding_datas
+
+
+def verify_embedding_data_vectors(original_embedding_datas, tracing_embedding_datas):
+    if len(original_embedding_datas) != len(tracing_embedding_datas):
+        print(
+            f"The length of original_embedding_data_vector: {len(original_embedding_datas)} and tracing_embedding_data_vector: {len(tracing_embedding_datas)} are different"
+        )
+        return False
+
+    for index, (original, tracing) in enumerate(
+        zip(original_embedding_datas, tracing_embedding_datas)
+    ):
+        if not verify_sparse_encoding(original, tracing):
+            print(
+                f"Verification failed for index {index}, whose input is {TEST_SENTENCES[index]}."
+            )
+            return False
+
+    return True
+
+
+def verify_sparse_encoding(
+    original_embedding_data: dict,
+    tracing_embedding_data: dict,
+) -> bool:
+    if original_embedding_data.keys() != tracing_embedding_data.keys():
+        print("Different encoding dimensions")
+        return False
+    tolerance = 0.01
+    for key in original_embedding_data:
+        if abs(original_embedding_data[key] - tracing_embedding_data[key]) > tolerance:
+            print(
+                f"{key}'s score has gap: {original_embedding_data[key]} != {tracing_embedding_data[key]}"
+            )
+            return False
+    return True
+
+
+def prepare_files_for_uploading(
+    model_id: str,
+    model_version: str,
+    model_format: str,
+    src_model_path: str,
+    src_model_config_path: str,
+) -> tuple[str, str]:
```

Comment:
Do you think reusing functions prepare_files_for_uploading from model_autotracing.py can work like this:
Create another base class with all the similar functions. Users can specify the model type as an argument